### ISSUE TITLE
Rule proposal: Prefer `array.flat()` over `array.flat(1)`

### ISSUE THREAD

**Issue by jonahsnider** [sindresorhus/eslint-plugin-unicorn#1629]
### Description

`Array.prototype.flat` uses `1` as the argument if you pass `undefined` or do not provide a value.
This means that `.flat(1)` is the same as `.flat()`, making the argument unnecessary.

### Fail

```js
array.flat(1);
```

```js
const one = 1;
array.flat(one);
```

### Pass

```js
array.flat();
```

```js
const one = 1;
array.flat();
```

```js
let levels = 1; // Mutable variable, might be reassigned later
array.flat(levels);
```

**Comment by bmish** [comment]
Would it make sense to just add this check to the existing prefer-array-flat rule?

**Comment by sindresorhus** [comment]
Not sure it belongs in `prefer-array-flat`. That rule is not about call style.

Would maybe be a better fit for https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1356?

**Comment by fisker** [comment]
We already have `require-array-join-separator` and `require-number-to-fixed-digits-argument` doing similar stuff, we can add this before we implement #1356.

**Comment by fisker** [comment]
I wonder if someone prefer use `array.flat(1)`,  it's explicit, if it does, we can add option for it.

**Comment by fregante** [comment]
All the rules mentioned here are a variation of "enforce or avoid the default value".

- avoids default: this suggestion
- enforces default: `require-array-join-separator`
- enforces default: `require-number-to-fixed-digits-argument`
- avoids default: `no-useless-undefined`

It almost feels like the belong together. I'm not particularly a fan of enforcing default values, but they do make it clearer sometimes.
